ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/ArithmeticInterpreter.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter;
+
+import static io.confluent.ksql.execution.interpreter.CastInterpreter.NumberConversions.toDouble;
+import static io.confluent.ksql.execution.interpreter.CastInterpreter.NumberConversions.toInteger;
+import static io.confluent.ksql.execution.interpreter.CastInterpreter.NumberConversions.toLong;
+
+import io.confluent.ksql.execution.expression.tree.ArithmeticUnaryExpression.Sign;
+import io.confluent.ksql.execution.interpreter.CastInterpreter.ConversionType;
+import io.confluent.ksql.execution.interpreter.terms.ArithmeticBinaryTerm;
+import io.confluent.ksql.execution.interpreter.terms.ArithmeticBinaryTerm.ArithmeticBinaryFunction;
+import io.confluent.ksql.execution.interpreter.terms.ArithmeticUnaryTerm;
+import io.confluent.ksql.execution.interpreter.terms.ArithmeticUnaryTerm.ArithmeticUnaryFunction;
+import io.confluent.ksql.execution.interpreter.terms.CastTerm;
+import io.confluent.ksql.execution.interpreter.terms.Term;
+import io.confluent.ksql.schema.Operator;
+import io.confluent.ksql.schema.ksql.types.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlDecimal;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.util.DecimalUtil;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+
+public final class ArithmeticInterpreter {
```

Comment:
how does arithmetic behave with nulls? would be good to add parity tests for this (or go down the QTT approach)

Reply:
I added null tests in `ExpressionEvaluatorParityTest`.  The code compiler actually doesn't handle a lot of null cases, though the interpreter does.  In some cases, the `ExpressionTypeManager` throws an error (or NPE often), and I didn't attempt to fix this just yet, so they both then have the same error.